freud/cluster.pyx

```diff
@@ -246,5 +309,30 @@ cdef class ClusterProperties(_Compute):
             &self.thisptr.getClusterSizes(),
             freud.util.arr_type_t.UNSIGNED_INT)
 
+    @_Compute._computed_property
+    def cluster_masses(self):
+        """(:math:`N_{clusters}`) :class:`numpy.ndarray`: The total mass of
+        particles in each cluster.
+        """
+        return freud.util.make_managed_numpy_array(
+            &self.thisptr.getClusterMasses(),
+            freud.util.arr_type_t.FLOAT)
+
+    @_Compute._computed_property
+    def radii_of_gyration(self):
+        r"""(:math:`N_{clusters}`,) :class:`numpy.ndarray`: The radius of
+        gyration of each cluster. Defined by IUPAP as
+
+        .. math::
+
+            R_g^k = \left(\frac{1}{M} \sum_{i=0}^{N_k} m_i s_i^2 \right)^{1/2}
+
+        where :math:`s_i` is the distance of particle :math:`i` from
+        the center of mass.
+
+        """
+        return np.sqrt(np.trace(self.inertia_tensors, axis1=-2, axis2=-1)
+                       /(2*self.cluster_masses))
```

Comment:
Let's keep this implementation that same as it was before (i.e. as the square root of the trace of the gyration tensor). The old way involved a little less number crunching.

Reply:
But to keep it equivalent to the IUPAP definition, we have to account for mass:
![image](https://user-images.githubusercontent.com/73320455/183964353-d7ce3fab-5704-4a57-9ba4-2d594f901ec7.png)
The use of square root of trace of inertia tensor for the actual calculation is given in the Appendix of this [paper](https://pubs.acs.org/doi/10.1021/jp2065612)

Reply:
I admit I'm a bit behind on in depth freud knowledge, but if there's a flag of some sort or any super efficient way to check if the system-like object has uniform mass, then it might make sense to have both implementations. Otherwise I'd maybe expect users to self-implement it if they really need a more performant version and stick to the most correct.

Reply:
Users can just give masses which are all 1 if they want the mass independent version, @AlainKadar is right, we will use the mass-dependent version.

Reply:
Honestly, my original comment was a bit nitpicky. There is slightly more number crunching this way vs the massless version, but the difference is slim to None. It is fine the way it is.